File: src/server/src/main/java/com/project/infrastructure/data/MuscleGroupRepository.java
package com.project.infrastructure.data;

import com.project.domain.workout.MuscleGroup;
import io.micronaut.data.annotation.Repository;
import io.micronaut.data.repository.CrudRepository;

import java.util.List;

@Repository
public interface MuscleGroupRepository extends CrudRepository<MuscleGroup, String> {
    List<MuscleGroup> findByIdIn(List<String> ids);
}

File: src/server/src/main/java/com/project/infrastructure/data/MuscleGroupRepository.kt
package com.project.infrastructure.data

import com.project.domain.workout.MuscleGroup
import io.micronaut.data.annotation.Repository
import io.micronaut.data.repository.CrudRepository

@Repository
interface MuscleGroupRepository : CrudRepository<MuscleGroup, String> {
    fun findByIdIn(ids: List<String>): List<MuscleGroup>
}
